client/my-sites/plugins/plugin-install-button/index.jsx

```diff
@@ -149,18 +152,27 @@ export class PluginInstallButton extends Component {
 	}
 
 	renderMarketplaceButton() {
-		const { translate, selectedSite, plugin, billingPeriod } = this.props;
+		const { translate, selectedSite, plugin, billingPeriod, canInstallPlugins } = this.props;
 		const variationPeriod = getPeriodVariationValue( billingPeriod );
 		const product_slug = plugin?.variations?.[ variationPeriod ]?.product_slug;
 
+		const buttonLink = canInstallPlugins
+			? `/checkout/${ selectedSite.slug }/${ product_slug }?redirect_to=/marketplace/thank-you/${ plugin.slug }/${ selectedSite.slug }`
+			: `/checkout/${ selectedSite.slug }/${ businessPlanToAdd(
+					selectedSite?.plan,
+					billingPeriod
+			  ) },${ product_slug }?redirect_to=/marketplace/thank-you/${ plugin.slug }/${
```

Comment:
To add the correct plan on every checkout, just like in `plugin-details-CTA/index.jsx`, I think we should add the `eligibleForProPlan` selector and param here:
```suggestion
			: `/checkout/${ selectedSite.slug }/${ businessPlanToAdd(
					selectedSite?.plan,
					billingPeriod,
					eligibleForProPlan
			  ) },${ product_slug }?redirect_to=/marketplace/thank-you/${ plugin.slug }/${
```

Reply:
You are right, we actually already solved this issue on trunk but as this PR is quite old it is not solved here yet. Thanks for pointing this.

Reply:
Fixed at https://github.com/Automattic/wp-calypso/pull/62353/commits/e6fbd01bb9d7c9e90e5eb195912cad33296ca790